src/app/shared/services/geojson.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GeojsonService {
+
+  constructor() { }
+
+  readFileContent(file: File) {
+    let fileReader: FileReader = new FileReader();
+    fileReader.readAsArrayBuffer(file);
+    return Observable.create(observer => {
+      fileReader.onloadend = () => {
+        observer.next(fileReader.result);
+        observer.complete();
+      };
+    });
+  }
```

Comment:
I wonder if this function can just be moved to a difference service file so we can delete this service file.

Reply:
Moved to loader.service, let me know if you think there is a better service to move it to.

Reply:
The loader service should just be for the loader. The app service is probably the best one.